pallets/price-oracle/src/types.rs

```diff
@@ -0,0 +1,215 @@
+// This file is part of Basilisk-node.
+
+// Copyright (C) 2020-2021  Intergalactic, Limited (GIB).
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use codec::{Decode, Encode};
+use frame_support::sp_runtime::traits::{CheckedDiv, Zero};
+use frame_support::sp_runtime::RuntimeDebug;
+pub use primitives::{Balance, Price};
+use sp_std::iter::Sum;
+use sp_std::ops::{Add, Index, IndexMut, Mul};
+use sp_std::prelude::*;
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(RuntimeDebug, Encode, Decode, Copy, Clone, PartialEq, Eq, Default)]
+pub struct PriceEntry {
+	pub price: Price,
+	pub amount: Balance,
+	pub liq_amount: Balance,
+}
+
+impl Add for PriceEntry {
+	type Output = Self;
+	fn add(self, other: Self) -> Self {
+		Self {
+			price: self.price.add(other.price),
+			amount: self.amount.add(other.amount),
+			liq_amount: self.liq_amount.add(other.liq_amount),
+		}
+	}
+}
+
+impl Zero for PriceEntry {
+	fn zero() -> Self {
+		Self {
+			price: Price::zero(),
+			amount: Balance::zero(),
+			liq_amount: Balance::zero(),
+		}
+	}
+
+	fn is_zero(&self) -> bool {
+		self == &PriceEntry::zero()
+	}
+}
+
+impl Add for &PriceEntry {
+	type Output = PriceEntry;
+	fn add(self, other: Self) -> Self::Output {
+		PriceEntry {
+			price: self.price.add(other.price),
+			amount: self.amount.add(other.amount),
+			liq_amount: self.liq_amount.add(other.liq_amount),
+		}
+	}
+}
+
+impl<'a> Sum<&'a Self> for PriceEntry {
+	fn sum<I>(iter: I) -> Self
+	where
+		I: Iterator<Item = &'a Self>,
+	{
+		iter.fold(
+			PriceEntry {
+				price: Price::zero(),
+				amount: Balance::zero(),
+				liq_amount: Balance::zero(),
+			},
+			|a, b| &a + b,
+		)
+	}
+}
+
+pub const BUCKET_SIZE: u32 = 10;
```

Comment:
This is probably going to be configured from time to time so might be better as a pallet const
Maybe NonZeroU32 type can be used instead?

Reply:
We can't use NonZeroU32 because its constructor is not `const`. The same problem applies also for pallet config parameters.